source/ambassador_host_test.go

```diff
@@ -141,3 +658,77 @@ func TestParseAmbLoadBalancerService(t *testing.T) {
 		}
 	}
 }
+
+func convertAmbassadorHostToUnstructured(hp *ambassador.Host, s *runtime.Scheme) (*unstructured.Unstructured, error) {
+	unstructuredAmbassadorHost := &unstructured.Unstructured{}
+	if err := s.Convert(hp, unstructuredAmbassadorHost, context.Background()); err != nil {
+		return nil, err
+	}
+	return unstructuredAmbassadorHost, nil
+}
+
+func newAmbassadorDynamicKubernetesClient() (*fakeDynamic.FakeDynamicClient, *runtime.Scheme) {
+	s := runtime.NewScheme()
+	_ = ambassador.AddToScheme(s)
+	return fakeDynamic.NewSimpleDynamicClient(s), s
+}
+
+type fakeAmbassadorHost struct {
```

Comment:
This overlaps with this function: https://github.com/kubernetes-sigs/external-dns/blob/master/source/ambassador_host_test.go#L86
Suggest refactor to use one or the other.